inspirehep/modules/records/cli.py

```diff
@@ -109,13 +120,15 @@ def simpleindex(yes_i_know, pid_type, batch_size, queue_name):
     click.secho('Sending record UUIDs to the indexing queue...', fg='green')
 
     query = (
-        db.session.query(PersistentIdentifier.object_uuid)
+        db.session.query(PersistentIdentifier.object_uuid).join(RecordMetadata, type_coerce(PersistentIdentifier.object_uuid, String) == type_coerce(RecordMetadata.id, String))
         .filter(
             PersistentIdentifier.pid_type.in_(pid_type),
             PersistentIdentifier.object_type == 'rec',
             PersistentIdentifier.status == PIDStatus.REGISTERED,
+            or_(RecordMetadata.json["deleted"].is_(None), RecordMetadata.json["deleted"] == cast(False, JSONB))
```

Comment:
* I don't think that's the right query. We never have `deleted=None`. You probably want something like `not_(type_coerce(RecordMetadata.json, JSONB).has_key('deleted'))` instead.
* Another issue is that this query will be very expensive, as we currently don't have an index on `deleted`, so it's a full table scan. As only a small fraction of records is deleted, wouldn't it be better to filter them out in Python instead?

Reply:
I added deleted None because I had checked previously and we do have deleted None and I also tested out the query and it works. 
Filtering them out in python sounds good though, I will look into it

Reply:
yes, you are right. If the deleted flag doesn't exist, then it returns None

Reply:
It's strange that there is no distinction between a `{"foo": null}` and `{}` in JSONB, but that seems to be the behavior here. I would still use the my suggestion as it expresses intent more clearly.

Reply:
>     * And I was thinking coercing whole record could take time, but no idea how these `JSON` blobs work.

`type_coercing` doesn't do anything with the data. It just tells SQLAlchemy that the thing you have in your hands actually has type `JSONB` contrarily to what it would have inferred (due to invenio also supporting different databases that don't have `JSONB`), so you can use `JSONB` methods on it.